test/e2e/specs/editor/plugins/meta-boxes.spec.js

```diff
@@ -0,0 +1,123 @@
+	} );
+
+	test( 'Should save the post', async ( { editor, page } ) => {
+		const saveDraft = page
+			.getByRole( 'region', { name: 'Editor top bar' } )
+			.getByRole( 'button', { name: 'Save draft' } );
+
+		// Save should not be an option for new empty post.
+		await expect( saveDraft ).toBeDisabled();
+
+		// Add title to enable valid non-empty post save.
+		await page
+			.getByRole( 'textbox', { name: 'Add title' } )
+			.fill( 'Hello Meta' );
+
+		await expect( saveDraft ).toBeEnabled();
+
+		await editor.saveDraft();
+
+		// After saving, affirm that the button returns to Save Draft.
+		await expect( saveDraft ).toBeEnabled();
+	} );
+
+	test( 'Should render dynamic blocks when the meta box uses the excerpt for front end rendering', async ( {
+		admin,
+		editor,
+		page,
+	} ) => {
+		// Publish a post so there's something for the latest posts dynamic block to render.
+		await page
+			.getByRole( 'textbox', { name: 'Add title' } )
+			.fill( 'A published post' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'Hello there!' );
+		await editor.publishPost();
+
+		// Publish a post with the latest posts dynamic block.
+		await admin.createNewPost();
+		await page
+			.getByRole( 'textbox', { name: 'Add title' } )
+			.fill( 'Dynamic block test' );
+		await editor.insertBlock( { name: 'core/latest-posts' } );
+
+		const postId = await editor.publishPost();
+		await page.goto( `/?p=${ postId }` );
+
+		await expect(
+			page.locator( '.wp-block-latest-posts > li' )
+		).toContainText( [ 'A published post', 'Dynamic block test' ] );
+	} );
+
+	test( 'Should render the excerpt in meta based on post content if no explicit excerpt exists', async ( {
+		editor,
+		page,
+	} ) => {
+		await page
+			.getByRole( 'textbox', { name: 'Add title' } )
+			.fill( 'A published post' );
+		await page.getByRole( 'button', { name: 'Add default block' } ).click();
+		await page.keyboard.type( 'Excerpt from content.' );
+
+		const postId = await editor.publishPost();
+		await page.goto( `/?p=${ postId }` );
+
+		await expect(
+			page.locator( 'meta[property="gutenberg:hello"]' )
+		).toHaveAttribute( 'content', 'Excerpt from content.' );
+	} );
+
+	test( 'Should render the explicitly set excerpt in meta instead of the content based one', async ( {
+		editor,
+		page,
+	} ) => {
+		await editor.openDocumentSettingsSidebar();
+		await page.getByRole( 'button', { name: 'Add default block' } ).click();
+		await page.keyboard.type( 'Excerpt from content.' );
+		await page
+			.getByRole( 'textbox', { name: 'Add title' } )
+			.fill( 'A published post' );
+
+		const documentSettings = page.getByRole( 'region', {
+			name: 'Editor settings',
+		} );
+		const excerptButton = documentSettings.getByRole( 'button', {
+			name: 'Excerpt',
+		} );
+
+		// eslint-disable-next-line playwright/no-conditional-in-test
+		if (
+			( await excerptButton.getAttribute( 'aria-expanded' ) ) === 'false'
+		) {
+			await excerptButton.click();
+		}
```

Comment:
Maybe we should turn off [`no-conditional-in-test`](https://github.com/playwright-community/eslint-plugin-playwright/blob/main/docs/rules/no-conditional-in-test.md) ESLint rule. Most of the warnings I've seen in our test are false positives, as we need to check the state of panels/sidebars before trying to open them.

Reply:
+1 on this. I haven't seen a case yet where this rule would be helpful.

Reply:
One question though! 😄 Is the excerptButton state actually unknown (or random?) at this point? Why do we need to check whether it's expanded or not?

Reply:
The panel state (excerptButton state) is persisted for a user and is sometimes shared across the e2e test shards. This can result in flaky tests.

Reply:
Gotcha. Definitely something to have as a side note there, then! 🙏 

Anyhow, I think we should avoid running parallel tests against the same instance since it makes the tests unpredictable (like this one) and thus unreliable. Do you think it would be possible to run separate instances or at least create separate admin accounts per shard? cc @kevin940726

Reply:
This is a known "quirk"; here's the previous discussion - #50851. Some of these details can be absorbed in editor helper utils, just like we do for `openDocumentSettingsSidebar`.

Reply:
> Do you think it would be possible to run separate instances or at least create separate admin accounts per shard?


What do you mean by "shard" here? If you mean playwright's `--shard` then I believe we are testing a different WordPress instance each shard.

Ideally, I think we should aim for fully parallel testing if possible, but the very nature of WordPress makes it very difficult to do. That's also why we still can't enable parallel mode to speed up the tests. Another gotcha as mentioned in #50851 is that the `resetPreferences()` call doesn't seem very reliable in tests. Perhaps there's something wrong in the API side or we are doing something wrong elsewhere. I never have time to investigate it but I should 😅. The goal is to make every test as isolated and predictable as possible with a clean state, so that we don't have to do some of those conditionals for most cases.

A bit of side-note here is that I had a brief (like literally 3-sentence) discussions with @adamziel about whether we can leverage the architecture of wp-playground somehow to automatically generate a unique WordPress instance (or a clean DB) for each test. This is a much more challenging task though and might not even work for performance reason 😆.

Reply:
@kevin940726 it might actually be possible now and perhaps without too much effort. Playground now supports the bits that were missing earlier like `proc_open()`, `system()` etc. functions which were crucial for the PHPUnit test suite. Also, almost all core unit tests work with the SQLite integration plugin. I didn't yet try running PHPUnit for WP Core with the new version of https://www.npmjs.com/package/@php-wasm/node, but there's a high likelihood it would just work.